L3/l3.scala

```diff
@@ -0,0 +1,116 @@
+// own list.reverse
+def listRev[A](list:List[A]):List[A]= {
+  def help(list:List[A],reversed:List[A]):List[A] = list match{
+    case Nil => reversed
+    case head::tail => help(tail,head::reversed)
+    }
+  help(list,Nil)
+}
+
+//1
+/*
+zlozonosc:
+obliczeniowa:O(n), n-dlugosc podanej listy
+pamieciowa:O(1),
+*/
+
+def checkNegative(n:Int):Boolean = {
+  n < 0
+}
+
+def checkOddNegative(n:Int):Boolean = {
+  (n < 0) && ((n%2)!=0 )
+}
+
+
+def splitList(list:List[Int]):(List[Int],List[Int]) = {
+  def help(list:List[Int],neg:List[Int],oddNeg:List[Int]):(List[Int],List[Int]) =list match {
+  case Nil => (listRev(neg),listRev(oddNeg))
+  case head::tail=>
+    if (checkOddNegative(head)) help(tail,head::neg,head::oddNeg)
```

Comment:
jeśli ujemne to pomóż... tak się czyta Twój program. Proszę nazywaj funkcje tak jak to co robią, bo potem jest wołanie o pomoc help() ;)